src/randomenv.cpp

```diff
@@ -44,15 +44,15 @@
 #if HAVE_DECL_GETIFADDRS && HAVE_DECL_FREEIFADDRS
 #include <ifaddrs.h>
 #endif
-#if HAVE_SYSCTL
+#ifdef HAVE_SYSCTL
```

Comment:
Would it be possible to change these to values instead of defined/undefined? It seems to me that changing these to `ifdef` makes it less safe because you lose the protection added by the warning in the first place. (say, when you forget to include bitcoin-config)

Reply:
> Would it be possible to change these to values 0/1 instead of defined/undefined?

These are currently checked via: `AC_CHECK_HEADERS([sys/select.h sys/prctl.h sys/sysctl.h vm/vm_param.h sys/vmmeter.h sys/resources.h])`. If we wanted, to we could change this.

Reply:
i still think it'd be better, but probably not worth doing this last-minute change for autotools.
Don't know how the CMake port handles these kind of checks anyway, but might consider it there.

Reply:
> but probably not worth doing this last-minute change for autotools.

Not sure. Everything not done for Autotools, but done in CMake is just more bebaviour/divergence to maintain, so I'd prefer to do as much as possible prior.